include/ball_sort/move.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <string>
+
+namespace ballsort {
+
+struct Move {
+    friend std::hash<Move>;
+
+    explicit Move(const std::pair<size_t, size_t> move,
+                  std::string serialised_state = "")
+        : m_origin{move.first}, m_destination{move.second},
+          m_serialised_state{std::move(serialised_state)}
+    {}
```

Comment:
You will probably want to have an overload that takes two `size_t`s to avoid the (minimal) overhead in constructing a pair